Markl update private sms tutorial

marklewin opened the pull request:
Make use of ES6 features and use Node promises instead of external dependency on `bluebird`

Commit:
Fix call to non-existent method SmsProxy.provisionLVNs

Commit:
Update with ES6 features

Commit:
Remove unused test route

mheap commented:
LGTM from a cursory read through, but I've not had chance to clone the repo and test it out. Could another reviewer test the app?

Commit:
Simplify app structure and config

Commit:
Simplify tutorial

Commit:
Simplify project layout

Commit:
Delete lib directory

Commit:
Regenerate package.json

Commit:
Change port 5000 -> 3000

Commit:
Update README

Commit:
Remove extraneous break statement

Commit:
Add starting example.env

marklewin commented:
Simplified tutorial code by removing everything that is not fundamental to the use case - such as Number Insight checking and automatic provisioning of virtual numbers.

mheap commented on example.env:
```diff
@@ -1,5 +1,4 @@
-PROVISIONED_NUMBERS=[{"country":"2_CHAR_CODE","msisdn":"NUMBER"}]
-SMS_WEBHOOK_URL=
+VIRTUAL_NUMBER_A=
+VIRTUAL_NUMBER_B=
```
Why do we need two virtual numbers here? We can link two numbers to a single number and then distribute the message based on the `from` field

marklewin replied:
Great idea! Refactored code to achieve this

mheap commented on server.js:
```diff
@@ -0,0 +1,50 @@
+    console.log('SMS Proxy App listening on port', app.get('port'));
+});
+
+const config = require('dotenv').config();
```
In most apps you don't assign `dotenv.config()` to anything. You just call `require('dotenv').config();` as the first line of your app and use `process.env.VARIABLE_NAME` where needed

marklewin replied:
Changed to use `process.env.VARIABLE_NAME`.

Commit:
Refactor to use single LVN

Commit:
Log and return if no chat found

Commit:
Tidy up chat object

lornajane committed:
Make incoming webhook accept GET, update README

Add more instructions to README, and get the `/chat` endpoint to return
something to avoid errors in the HTTP client. Switch the incoming
webhook to GET as this is more common (if not exactly better)
configuration and is account-wide so changing it for this application
could stop other applications from working.

lornajane commented:
@marklewin As discussed I've flipped the incoming webhook to be GET rather than POST, simply because it's an account-wide setting and is commonly GET. This is a great update and if you're happy, I'll merge

lornajane reviewed (approved):
Much simpler, thanks!

marklewin commented:
Thanks for this. Looks great!

Commit:
Improve comments in sendSMS()
